File: WebApplication1/ClassLibrary1/Models/BooksModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLibrary1.Models
{
    public class BooksModel
    {
        public int BookId { get; set; }
        public string BookName { get; set; }
        public string PublisherName { get; set; }
        public string DepartmentName { get; set; }
        public string SupplierName { get; set; }
        public bool IsActive { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime ModifiedOn { get; set; }

   
    }
}

File: LibraryManagementSystem/src/Models/BooksModel.ts
import { PublishersModel } from "./PublihsersModel"; 

export class BooksModel {
    BookNameSearch?: string = '';
    SelectedPublisher: string='';
    PageNo: number = 1;
    PageSize: number = 5;
    SortColumn?: string='';
    SortOrder?: string='';
    SelectedIsActive?: number=-1;
    SelectedPublishers?: PublishersModel[];





    BookId : number=0;
    BookName? : string='';
    PublisherId : number | null =null;
    PublisherName? : string='';
    DepartmentId : number | null=null;
    DepartmentName? : string='';
    SupplierId : number | null=null;
    SupplierName? : string='';
    NoOfCopies : number | null =null ;
    IsActive : Boolean= false;
    CreatedBy : number | null  = null ;
  }